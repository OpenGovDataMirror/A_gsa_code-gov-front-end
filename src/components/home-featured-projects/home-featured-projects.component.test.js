import React from 'react'
import { shallow } from 'enzyme'

import HomeFeaturedProject from 'components/home-featured-project/home-featured-project.component'
import HomeFeaturedProjects from 'components/home-featured-projects/home-featured-projects.component'

const props = {
  featuredProjects: [
    { short_name: 'project-1' },
    { short_name: 'project-2' },
  ],
}

let wrapper
describe('components - HomeFeaturedProjects', () => {
  beforeEach(() => {
    wrapper = shallow(<HomeFeaturedProjects {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should map a list of `HomeFeaturedProject`s based off the `featuredProjects` props', () => {
    expect(wrapper.find(HomeFeaturedProject).length).toBe(props.featuredProjects.length)
  })

  it('should not display any `HomeFeaturedProject`s if not supplied any `featuredProjects`', () => {
    wrapper.setProps({ featuredProjects: undefined })
    expect(wrapper.find(HomeFeaturedProject).length).toBe(0)
  })
})