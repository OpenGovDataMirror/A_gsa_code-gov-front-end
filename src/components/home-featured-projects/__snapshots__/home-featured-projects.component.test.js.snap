// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - HomeFeaturedProjects should render correctly 1`] = `
<section
  className="featured-projects block block--white"
>
  <div
    id="featured-projects-title"
  >
    <div
      className="indented"
    >
      <h2>
        Featured Projects
      </h2>
    </div>
  </div>
  <div>
    <HomeFeaturedProject
      index={0}
      key="project-1"
      project={
        Object {
          "short_name": "project-1",
        }
      }
    />
    <HomeFeaturedProject
      index={1}
      key="project-2"
      project={
        Object {
          "short_name": "project-2",
        }
      }
    />
  </div>
</section>
`
